packages/connect-core/src/entities/Intent.ts

```diff
@@ -0,0 +1,81 @@
+import { ethers } from 'ethers'
+
+import Organization from './Organization'
+import Transaction from './Transaction'
+import TransactionPath from '../transactions/TransactionPath'
+import { PathOptions, IntentData } from '../types'
+import { calculateTransactionPath } from '../utils/path/calculatePath'
+
+export default class Intent {
+  #org: Organization
+  #provider: ethers.providers.Provider
+  readonly appAddress: string
+  readonly functionName: string
+  readonly functionArgs: any[]
+
+  constructor(
+    data: IntentData,
+    org: Organization,
+    provider: ethers.providers.Provider
+  ) {
+    this.#org = org
+    this.#provider = provider
+    this.appAddress = data.appAddress
+    this.functionName = data.functionName
+    this.functionArgs = data.functionArgs
+  }
+
+  // Retrieve a single forwarding path. Defaults to the shortest one.
+  async path({ actAs, path }: PathOptions): Promise<TransactionPath> {
```

Comment:
We removed all the logic related to the description of a path and handle that responsibility to the Forwarding Path.